src/remote_write/src/pooled_parser.rs

```diff
@@ -0,0 +1,221 @@
+// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+//! Pooled parser for Prometheus remote write requests.
+//!
+//! This crate parses the protobuf `string` type as Rust `Bytes` instances
+//! instead of `String` instances to avoid allocation, and it **does not**
+//! perform UTF-8 validation when parsing. Therefore, it is up to the caller to
+//! decide how to make use of the parsed `Bytes` and whether to apply UTF-8
+//! validation.
+//!
+//! ## Basic Usage
```

Comment:
I prefer we remove all comments below L26 , the API is not stable now, and very likely the comments become out of date when we refactor the API.